src/foundry/client-esm/applications/api/application.d.mts

```diff
@@ -556,40 +586,31 @@ declare class ApplicationV2<
    * @param context - Prepared context data
    * @param options - Provided render options
    */
-  protected _preFirstRender(
-    context: DeepPartial<ApplicationV2.RenderContext>,
-    options: DeepPartial<RenderOptions>,
-  ): Promise<void>;
+  protected _preFirstRender(context: DeepPartial<RenderContext>, options: DeepPartial<RenderOptions>): Promise<void>;
```

Comment:
This is why I had to update the mixin; `ApplictionV2.RenderContext` is extremely permissive and in reality this has to match whatever the subclass defines.